sktime/forecasting/model_evaluation/_functions.py

```diff
@@ -471,51 +478,34 @@ def gen_y_X_train_test(y, X, cv, cv_X):
     # generator for y and X splits to iterate over below
     yx_splits = gen_y_X_train_test(y, X, cv, cv_X)
 
+    # sequential strategies cannot be parallelized
+    not_parallel = strategy in ["update", "no-update_params"]
+
     # dispatch by backend and strategy
-    if backend is None or strategy in ["update", "no-update_params"]:
+    if not_parallel:
         # Run temporal cross-validation sequentially
         results = []
-        for i, (y_train, y_test, X_train, X_test) in enumerate(yx_splits):
-            if strategy == "update" or (strategy == "no-update_params" and i == 0):
-                result, forecaster = _evaluate_window(
-                    y_train,
-                    y_test,
-                    X_train,
-                    X_test,
-                    i,
-                    **_evaluate_window_kwargs,
-                )
+        for x in enumerate(yx_splits):
```

Comment:
not a bloking comment but would appreciate it if `x`  is actually being decomposed to their own respective components like the status quo for readability. had some difficulty in knowing what `x` was just know.

Reply:
but then the abstraction wouldn't work? Or, how do you propose that it works in *both* refactor target cases?

Reply:
oh yeah, you are right! It wouldn't work.